aptos-labs/aptos-core
# [move-compiler-v2] reorganize/simplify options in aptos-move CLI, adding --lint and --optimization-level={none,standard,full} flags; also implement aptos move lint

## Description

Various CLI changes to support Move 2
- Reorganize and simplify option handling in `aptos move` cli a bit
- Add `aptos move lint` tool that just runs lint pass, exiting after extended_checks.
- Slightly improve `--experiments` passage from Aptos CLI, to allow `=on/off`
- Add `--optimization-level=none/standard/full` flag, passing it to compiler-v2 using `experiments` to avoid extra plumbing.

After some discussion, I've 
- Removed the `--lint` flag from `aptos move build`.
- Separated the `aptos move lint` options from `aptos move build` to make the tool simpler and avoid confusing options which aren't useful for lint.

Note that this could be improved:
- experiments could be multi-valued options, rather than just binary options
- move_options could be further simplified and shared across more modules,
  but this is enough for now

Fixes #14355: add --lint flag to CLI.
Fixes #14327: add flag --optimize=level to CLI.

## How Has This Been Tested?

`move_pr -i2`, the usual tests

Manual test of aptos cli:
- `aptos move lint` and `aptos build --lint` on a sample package containing lint test inputs
- `aptos move build --optimization-level=standard/full/none` and manual examination of some
  output binary files to ensure different outputs for different levels
  - on `aptos-move/framework/move-stdlib`
  - on a sample project with Wolfgang's enum-test cases to validate Move 2 support
- `aptos move build --help` and `aptos move lint --help` and careful scrutiny of outputs.

## Key Areas to Review

I added a few experiment hooks for different places to stop the
compiler for `aptos move lint`; currently it's
`STOP_AFTER_EXTENDED_CHECKS`, as that includes all normal
compiler warnings/errors in addition to the lint warnings.

## Type of Change

## Which Components or Systems Does This Change Impact?

## Checklist
